go/vt/vttablet/tabletmanager/vreplication/controller.go

```diff
@@ -270,18 +278,16 @@ func (ct *controller) runBlp(ctx context.Context) (err error) {
 
 		vr := newVReplicator(ct.id, ct.source, vsClient, ct.blpStats, dbClient, ct.mysqld, ct.vre)
 		err = vr.Replicate(ctx)
-		if isUnrecoverableError(err) {
-			settings, _, errSetting := vr.readSettings(ctx)
-			if errSetting != nil {
-				return err // yes, err and not errSetting.
-			}
-			if settings.WorkflowType == int64(binlogdatapb.VReplicationWorkflowType_ONLINEDDL) {
-				// Specific to OnlineDDL, if we encounter an "unrecoverable error", we change the migration state into Error and then we quit the workflow
-				if errSetState := vr.setState(binlogplayer.BlpError, err.Error()); errSetState != nil {
-					return err // yes, err and not errSetState.
-				}
-				return nil // this will cause vreplicate to quit the workflow
+
+		ct.lastWorkflowError.record(err)
+		if isUnrecoverableError(err) /* mysql error that we know needs manual intervention */ ||
+			!ct.lastWorkflowError.canRetry() /* cannot detect if this is recoverable, but it is persisting too long */ {
```

Comment:
Let's move `{` before the code comment for better clarity
```suggestion
			!ct.lastWorkflowError.canRetry() { // cannot detect if this is recoverable, but it is persisting too long
```